tests/neg/summonFrom-implicit-not-found.scala

```diff
@@ -0,0 +1,9 @@
+@annotation.implicitNotFound("there is no Missing!")
+trait Missing
+
+inline def summonMissing = compiletime.summonFrom {
+  case m: Missing => m
+}
```

Comment:
I wonder what should be the error for  
```scala
@annotation.implicitNotFound("there is no Missing1!")
trait Missing1
@annotation.implicitNotFound("there is no Missing2!")
trait Missing2
inline def summonMissing = compiletime.summonFrom {
  case m: Missing1 => m
  case m: Missing2 => m
}
val x = summonMissing

We can choose the first one, last one, concatenate the messages. Not sure what would be best.

Reply:
In that case it might be better to do

```scala
inline def summonMissing = compiletime.summonFrom {
  case m: Missing1 => m
  case m: Missing2 => m
  case _ => compiletime.error("no Missing1 or Missing2 found")
}
```

Reply:
Currently I think that we should use `compiletime.error` over `implicitNotFound`. The error messages in the `implicitNotFound` usually assume that this is the only implicit beeing summoned. Reusing `implicitNotFound` might lead to error messages that are out of context.

Reply:
On the other hand, proposing the first one or last one would at least solve the compile error at the sacrifice of information loss.

Reply:
Should we report unexhaustiveness as error in summonFrom with an error message better than "cannot reduce summonFrom with ..." in the similar way as we warn on unexhaustive pattern matches to enums?
(e.g. This does not compile because no instance of Missing1 or Missing2 is found.)

Reply:
Reminiscent of implicitNotFound on parent classes. IIRC scala 2 appends (accumulates); there is an open dotty ticket.

Reply:
If you think it is better to keep `unreduced summon from...` to prevent users from getting confused, I will close this PR. Otherwise, I will add changes to accumulate error messages from implicitsNotFound annotations.

Reply:
Thank you for the feedback. I'll address this.